### PR TITLE
refactor docs

### PR DESCRIPTION
**Pull request by njzjz**
See https://explore-deepmd.njzjz.win/ for preview.
PDF: https://explore-deepmd.njzjz.win/_/downloads/en/latest/pdf/

### CODE REVIEW

File: doc/getting-started/training.md [hunk 1]
```diff
@@ -0,0 +1 @@
+../train/training.md
```
Comment (amcadmus) [inline doc/getting-started/training.md:1]: Too much advanced topic in this training.md. 

Guide the user with a simple example and explain the out put in the `lcurve.out`

File: doc/install/install-from-source.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  -h, --help           show this help message and exit
+
+Valid subcommands:
+  {train,freeze,test}
+    train              train a model
+    freeze             freeze the model
+    test               test the model
+```
+
+### Install the C++ interface
```
Comment (amcadmus) [inline doc/install/install-from-source.md:95]: Shouldn't be `## Install the C++ interface `?

**Review by amcadmus — CHANGES_REQUESTED** [review]
0. The links in the `README.md` at the deepmd-kit source root are broken.
1. How a user conveniently read the doc locally or from the github website?  or we only want to provide the doc through readthedoc?

**Comment by njzjz** [comment]
> How a user conveniently read the doc locally or from the github website? or we only want to provide the doc through readthedoc?

It's hard to support all of them at the same time. However, I can provide a `README` under `doc` directory to give a guide how to build the website locally. For GitHub, I think one can easily visit our website if he can visit the GitHub?

File: doc/getting-started/training.md (hunk 1 above)
Comment (njzjz) [reply doc/getting-started/training.md:1]: Going to update tomorrow. It's too late in my time zone.

File: doc/model/train-energy.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+```json
+	"fitting_net" : {
+	   	"type": "polar",
+		"sel_type": [0],
+		"neuron": [100,100,100],
+		"resnet_dt": true,
+		"seed": 1,
+	},
+```
```
Comment (amcadmus) [inline doc/model/train-energy.md:29]: They are fitting nets for tonsorial properties

File: README.md [hunk 4]
```diff
@@ -51,29 +51,67 @@ In addition to building up potential energy models, DeePMD-kit can also be used
...
+    - [Data conversion](doc/data-conv.md)
+    - [Prepare data with dpdata](doc/dpdata.md)
+- [Model](doc/model/index.rst)
+    - [Overall](doc/model/overall.md)
+    - [Descriptor `"se_e2_a"`](doc/model/train-se-e2-a.md)
+    - [Descriptor `"se_e2_r"`](doc/model/train-se-e2-r.md)
+    - [Descriptor `"se_e3"`](doc/model/train-se-e3.md)
+    - [Descriptor `"hybrid"`](doc/model/train-hybrid.md)
+	- [Fit energy](doc/train-energy.md)
+    - [Fit `tensor` like `Dipole` and `Polarizability`](doc/train-fitting-tensor.md)
```
Comment (amcadmus) [inline README.md:91]: broken links

File: doc/model/train-se-e2-a.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+```json
+	"fitting_net" : {
+	    "neuron":		[240, 240, 240],
+	    "resnet_dt":	true,
+	    "seed":		1
+	},
+```
+* `neuron` specifies the size of the fitting net. If two neighboring layers are of the same size, then a [ResNet architecture](https://arxiv.org/abs/1512.03385) is built between them. 
+* If the option `resnet_dt` is set `true`, then a timestep is used in the ResNet. 
+* `seed` gives the random seed that is used to generate random numbers when initializing the model parameters.
```
Comment (amcadmus) [inline doc/model/train-se-e2-a.md:52]: mv this part to model/fitting-energy?

File: doc/model/train-se-e2-a.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    }
+```
+* `start_lr` gives the learning rate at the beginning of the training.
+* `stop_lr` gives the learning rate at the end of the training. It should be small enough to ensure that the network parameters satisfactorily converge. 
+* During the training, the learning rate decays exponentially from `start_lr` to `stop_lr` following the formula.
+    ```
+    lr(t) = start_lr * decay_rate ^ ( t / decay_steps )
+    ```
+    where `t` is the training step.
+
```
Comment (amcadmus) [inline doc/model/train-se-e2-a.md:73]: Also mv this part to the advance options of training paramters?

File: doc/model/train-energy.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
+# Fit energy
+
```
Comment (amcadmus) [inline doc/model/train-energy.md:2]: Add position of the example: `example/water/se_e2_a/input.json`, where reader can find the sections "fitting" and "loss"

**Review by amcadmus — CHANGES_REQUESTED** [review]

File: doc/model/train-energy.md (hunk 3 above)
Comment (njzjz) [reply doc/model/train-energy.md]: Fixed.

File: README.md (hunk 4 above)
Comment (njzjz) [reply README.md]: Fixed.

File: doc/model/train-se-e2-a.md (hunk 5 above)
Comment (njzjz) [reply doc/model/train-se-e2-a.md]: Resolved.

File: doc/model/train-se-e2-a.md (hunk 6 above)
Comment (njzjz) [reply doc/model/train-se-e2-a.md]: Resolved.

File: doc/model/train-energy.md (hunk 7 above)
Comment (njzjz) [reply doc/model/train-energy.md:2]: Resolved.

File: doc/train/training-advanced.md [hunk 8]
```diff
@@ -1,5 +1,67 @@
...
+	    "numb_btch":	3
+	},
+
+	"numb_step":	1000000,
+	"seed":		1,
+	"disp_file":	"lcurve.out",
+	"disp_freq":	100,
+	"save_freq":	1000
+    }
+```
```
[the remark this answers is not in the document]
Comment (njzjz) [reply doc/train/training-advanced.md:46]: Resolved.

**Review by amcadmus — APPROVED** [review]

**Review by tuoping — CHANGES_REQUESTED** [review]
You can change this: 
```python
exhale_args = {
   ...
    "rootFileName":          "api_cc.rst",
   ...
}
```
to
```python
exhale_args = {
   ...
    "rootFileName":          "../development/api_cc.rst",
   ...
}
```
Then the automatically built api_cc.rst will be in directory doc/development/

File: doc/index.rst [hunk 9]
```diff
@@ -11,41 +11,47 @@ DeePMD-kit is a package written in Python/C++, designed to minimize the effort r
...
 
 .. _developer-guide:
 
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 3
    :caption: Developer Guide
    
-   development/index
+   api
```
Comment (tuoping) [inline doc/index.rst:45]: And you can change here to
```python
.. toctree
    :maxdepth: 3
    :caption: Developer Guide
    :glob:

    development/*
```

File: doc/troubleshooting/index.rst [hunk 10]
```diff
@@ -0,0 +1,27 @@
+FAQs
+====
+In consequence of various differences of computers or systems, problems may occur. Some common circumstances are listed as follows. 
+In addition, some frequently asked questions about parameters setting are listed as follows.
+If other unexpected problems occur, you're welcome to contact us for help.
+
+.. _trouble:
+
+.. toctree::
```
Comment (tuoping) [inline doc/troubleshooting/index.rst:9]: For this directory to handle possible changes in the future, may we consider this:
```python
.. toctree::
    :maxdepth: 1
    :caption: trouble shooting
    :glob:

    ./*
```

**Review by tuoping — COMMENTED** [review]
Also, regarding @amcadmus 's comments, you can have both index.rst and index.md in the same directory, RST will go for index.rst automatically.

File: doc/index.rst (hunk 9 above)
Comment (njzjz) [reply doc/index.rst:45]: But, sections will be unordered?

Comment (tuoping) [reply doc/index.rst:45]: Yes, but I think "trouble shooting" is a problem pool for people to mine in, so sections may not be necessary?

Comment (tuoping) [reply doc/index.rst:45]: The sections won't change much. Originally, the list was like this 
```python
.. toctree::
   :maxdepth: 3
   :caption: Developer Guide

   api
   API_CC/api_cc
   development/coding-conventions
   development/type-embedding
```
The automatically generated one with ":glob:" is exactly the same.

File: doc/troubleshooting/index.rst (hunk 10 above)
Comment (tuoping) [reply doc/troubleshooting/index.rst:9]: The sections in "doc/troubleshooting" will be ordered alphebatically. And the subsections will be lost. About this, I think "troubleshooting" is a problem pool for people to mine in, so subsections may not be necessary.

**Comment by njzjz** [comment]
> You can change this:
> 
> ```python
> exhale_args = {
>    ...
>     "rootFileName":          "api_cc.rst",
>    ...
> }
> ```
> 
> to
> 
> ```python
> exhale_args = {
>    ...
>     "rootFileName":          "../development/api_cc.rst",
>    ...
> }
> ```
> 
> Then the automatically built api_cc.rst will be in directory doc/development/

Not work. `api_cc.rst` relies other files.

File: doc/index.rst [hunk 11]
```diff
@@ -11,41 +11,47 @@ DeePMD-kit is a package written in Python/C++, designed to minimize the effort r
...
 
 .. _developer-guide:
 
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 3
    :caption: Developer Guide
    
-   development/index
+   development/api
```
Comment (tuoping) [inline doc/index.rst:45]: This works:
```rst
.. toctree::
   :maxdepth: 3
   :caption: Developer Guide
   :glob:

   development/*
   API_CC/api_cc
```

Comment (tuoping) [reply doc/index.rst:45]: The sections will be ordered like this:
```html
Developer Guide

Python API
Coding Conventions
Atom Type Embedding
C++ API
```

File: doc/troubleshooting/index.rst (hunk 10 above)
Comment (tuoping) [reply doc/troubleshooting/index.rst:9]: This turned out to work out fine.

File: doc/index.rst (hunk 11 above)
Comment (tuoping) [reply doc/index.rst:45]: Preview here:
https://deepmd-kit.readthedocs.io/en/refactor-docs/

**Review by tuoping — APPROVED** [review]

**Pull request merged**